PWGCF/FemtoWorld/Core/FemtoWorldUtils.h

```diff
@@ -0,0 +1,123 @@
+/// \param nSigma number of sigma selection fo PID
+/// \param vNsigma vector with available n-sigma selections for PID
+/// \param kDetector enum corresponding to the PID technique
+/// \return Whether the PID selection specified in the vectors is fulfilled
+bool isPIDSelected(aod::femtodreamparticle::cutContainerType const& pidcut, std::vector<int> const& vSpecies, int nSpecies, float nSigma, const std::vector<float>& vNsigma, const kDetector iDet = kDetector::kTPC)
+{
+  bool pidSelection = true;
+  kPIDselection iNsigma = getPIDselection(nSigma, vNsigma);
+  for (auto iSpecies : vSpecies) {
+    //\todo we also need the possibility to specify whether the bit is true/false ->std>>vector<std::pair<int, int>>
+    // if (!((pidcut >> it.first) & it.second)) {
+    int bit_to_check = nSpecies * kDetector::kNdetectors * iNsigma + iSpecies * kDetector::kNdetectors + iDet;
+    if (!(pidcut & (1UL << bit_to_check))) {
+      pidSelection = false;
+    }
+  }
+  return pidSelection;
+};
+
+/// function that checks whether the PID selection specified in the vectors is fulfilled, depending on the momentum TPC or TPC+TOF PID is conducted
+/// \param pidcut Bit-wise container for the PID
+/// \param momentum Momentum of the track
+/// \param pidThresh Momentum threshold that separates between TPC and TPC+TOF PID
+/// \param vSpecies Vector with the species of interest (number returned by the CutCulator)
+/// \param nSpecies number of available selected species (output from cutculator)
+/// \param nSigmaTPC Number of TPC sigmas for selection
+/// \param nSigmaTPCTOF Number of TPC+TOF sigmas for selection (circular selection)
+/// \return Whether the PID selection is fulfilled
+bool isFullPIDSelected(aod::femtodreamparticle::cutContainerType const& pidCut, float const momentum, float const pidThresh, std::vector<int> const& vSpecies, int nSpecies, const std::vector<float>& vNsigma = {3.5, 3., 2.5}, const float nSigmaTPC = 3.5, const float nSigmaTPCTOF = 3.5)
+{
+  bool pidSelection = true;
+  if (momentum < pidThresh) {
+    /// TPC PID only
+    pidSelection = isPIDSelected(pidCut, vSpecies, nSpecies, nSigmaTPC, vNsigma, kDetector::kTPC);
+  } else {
+    /// TPC + TOF PID
+    pidSelection = isPIDSelected(pidCut, vSpecies, nSpecies, nSigmaTPCTOF, vNsigma, kDetector::kTPCTOF);
+  }
+  return pidSelection;
+};
+
+/// function to retrieve the nominal mgnetic field in kG (0.1T) and convert it directly to T
+/// \param timestamp timestamp corresponding the the collision
+/// \param ccdb CCDB object that contains the configuration for the event
+/// \return magnetic field in Tesla
+
+float getMagneticFieldTesla(const uint64_t& timestamp, const Service<o2::ccdb::BasicCCDBManager>& ccdb)
+{
+  // TODO done only once (and not per run). Will be replaced by CCDBConfigurable
+  static o2::parameters::GRPObject* grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>("GLO/GRP/GRP", timestamp);
```

Comment:
Have in mind that for Run 3 the `CCDB` endpoint and the way of getting the magnetic field information has changed
Have a look at the implementation of PR#976 https://github.com/AliceO2Group/O2Physics/pull/976/files for inspiration
If you are not addressing Run 3 datasets for the time being you don't need to consider it. Just have it in mind